includes/course-theme/class-sensei-course-theme-quiz.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * File containing Sensei_Course_Theme_Quiz class.
+ *
+ * @package sensei-lms
+ * @since 3.15.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Sensei_Course_Theme_Quiz class.
+ */
+class Sensei_Course_Theme_Quiz {
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_Course_Theme_Quiz constructor. Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		if ( 'quiz' !== get_post_type() ) {
+			return;
+		}
+
+		$this->maybe_add_quiz_results_notice();
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function maybe_add_quiz_results_notice() {
+
+		$lesson_id   = Sensei_Utils::get_current_lesson();
+		$quiz_id     = \Sensei()->lesson->lesson_quizzes( $lesson_id );
+		$user_id     = get_current_user_id();
+		$quiz_status = Sensei_Utils::user_lesson_status( $lesson_id, $user_id )->comment_approved;
+
+		// If not one of the statuses that we handle then bail.
```

Comment:
I think now we should tweak this comment too. Maybe "If not submitted"?

Reply:
Yup. Forgot about that. Updated. cadcc75ce19908282a81694a786eb35dbcf5d853